lib/bridge/MatrixHandler.js

```diff
@@ -770,6 +769,68 @@ MatrixHandler.prototype._onMessage = Promise.coroutine(function*(req, event) {
     yield Promise.all(promises);
 });
 
+MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
+    function*(req, ircRoom, ircUser, ircAction, event) {
+    let self = this;
+
+    if (event.content.msgtype !== "m.text" ||
+        !(ircUser.unsafeClient && ircUser.unsafeClient.wouldSend)) {
+        yield self.ircBridge.sendIrcAction(ircRoom, ircUser, ircAction);
+    }
+    else {
+        let text = event.content.body;
+        let wouldSend = ircUser.unsafeClient.wouldSend(ircRoom.channel, text);
+        let lineLimit = 2;
+        if (wouldSend.length > lineLimit) {
+            // Message body too long - send a file to HS instead and send link to IRC
+
+            try {
+                // Upload as a file and get URI
+
+                let fileName = Date().replace(/\s/g, '_') + ".txt";
```

Comment:
It would be polite to state what you're trying to do here e.g:

```javascript
// Use the current time as the name of the file and remove whitespace
let fileName = Date().replace(/\s/g, '_') + ".txt";
```